src/plugins/data/public/ui/dataset_selector/configurator.test.tsx

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { Configurator } from './configurator'; // Adjust the import path as needed
+import '@testing-library/jest-dom';
+import React from 'react';
+import { setQueryService, setIndexPatterns } from '../../services';
+import { IntlProvider } from 'react-intl';
+
+// Mocking the external service and hook functions
+const mockGetQueryService = jest.fn();
+const mockGetIndexPatterns = jest.fn();
+const mockQueryService = {
+  queryString: {
+    getLanguageService: jest.fn().mockReturnValue({
+      getLanguage: jest.fn().mockImplementation((langId) => ({
+        title:
+          {
+            en: 'English',
+            fr: 'French',
```

Comment:
I don't think this is the "language" `getLanguageService` provides. @kavilla am i wrong?

Reply:
Lets combine the 2 Pr's so that they are contextually together. Want to ensure that the tests are covering the right functionality

Reply:
addressed